lib/sdk/adapters/index.js

```diff
@@ -28,10 +31,8 @@ const ADAPTERS = [
   require('./blockdevice')
 ]
 
-// We don't support usbboot on GNU/Linux yet, given
-// that some distributions require root permissions
-// to open USB devices.
-if (os.platform() !== 'linux') {
+// On GNU/Linux, we only support usbboot when running as root.
+if ((os.platform() !== 'linux') || (process.geteuid() === permissions.UNIX_SUPERUSER_USER_ID)) {
```

Comment:
Isn't this duplicating the functionality already contained inside `permissions.isElevated()` ? Or are you doing it this way simply to avoid using Promises?

Reply:
We can't use isElevated here as it returns a promise and we want to set `exports` __now__.